### PR TITLE
Base+LibGUI: Add insert emoji common action

### PR DESCRIPTION
**Pull request by electrikmilk**
This adds a common action to invoke the emoji picker.

This adds the menu item to Text Editor where I think it seems most appropriate.

Screen Shot 2022-09-07 at 18 36 29

**Demo:**

https://user-images.githubusercontent.com/4368524/189005241-823a1fd2-851a-42ed-a642-4ab6fbdfa972.mov

### CODE REVIEW

File: Userland/Applications/TextEditor/MainWidget.cpp [hunk 1]
```diff
@@ -366,6 +366,7 @@ void MainWidget::initialize_menubar(GUI::Window& window)
     edit_menu.add_action(m_editor->copy_action());
     edit_menu.add_action(m_editor->paste_action());
     edit_menu.add_separator();
+    edit_menu.add_action(m_editor->insert_emoji_action());
```
Comment (linusg) [inline Userland/Applications/TextEditor/MainWidget.cpp:369]: I think this should be in a separate `TextEditor:` commit (and maybe we can find a few more places to enable it?)

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 2]
```diff
@@ -105,6 +106,8 @@ void TextEditor::create_actions()
             this);
     }
     m_select_all_action = CommonActions::make_select_all_action([this](auto&) { select_all(); }, this);
+    m_insert_emoji_action = CommonActions::make_insert_emoji_action([&](auto&) { insert_emoji(); },
+        this);
```
Comment (linusg) [inline Userland/Libraries/LibGUI/TextEditor.cpp:110]: Why `this` on a new line?

Comment (electrikmilk) [reply Userland/Libraries/LibGUI/TextEditor.cpp:110]: That was formatting by CLion before I moved the contents of `insert_emoji()` into its own function, my bad.

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 3]
```diff
@@ -771,6 +773,17 @@ void TextEditor::select_all()
...
+void TextEditor::insert_emoji()
+{
+    auto emoji_input_dialog = EmojiInputDialog::construct(window());
+    emoji_input_dialog->set_window_mode(GUI::WindowMode::Passive);
+    if (emoji_input_dialog->exec() != EmojiInputDialog::ExecResult::OK)
+        return;
+
+    auto emoji_code_point = emoji_input_dialog->selected_emoji_text();
+    insert_at_cursor_or_replace_selection(emoji_code_point);
+}
```
[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibGUI/TextEditor.cpp:785]: wait no, it doesn't!

[the remark this answers is not in the document]
Comment (trflynn89) [reply Userland/Libraries/LibGUI/TextEditor.cpp:785]: ...doesn't it?

[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibGUI/TextEditor.cpp:785]: I got confused between TextEditor, the LibGUI component, and TextEditor, the application (this is the former).

[the remark this answers is not in the document]
Comment (trflynn89) [reply Userland/Libraries/LibGUI/TextEditor.cpp:785]: :man_facepalming: same!

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 4]
```diff
@@ -105,6 +106,7 @@ void TextEditor::create_actions()
             this);
     }
     m_select_all_action = CommonActions::make_select_all_action([this](auto&) { select_all(); }, this);
+    m_insert_emoji_action = CommonActions::make_insert_emoji_action([&](auto&) { insert_emoji(); }, this);
```
Comment (trflynn89) [inline Userland/Libraries/LibGUI/TextEditor.cpp:109]: I think this action needs to be disabled in `TextEditor::set_mode`. We toggle `set_accepts_emoji_input` in there already.

Comment (electrikmilk) [reply Userland/Libraries/LibGUI/TextEditor.cpp:109]: Good catch!

**Pull request merged**